locals.tf

```diff
@@ -5,86 +5,420 @@ locals {
+      schema_grants = [{
+        privileges              = ["USAGE"]
+        all_schemas_in_database = true
+      }]
+      schema_objects_grants = {
+        for object_type in ["TABLE", "DYNAMIC TABLE", "EXTERNAL TABLE", "VIEW", "MATERIALIZED VIEW", "FILE FORMAT", "FUNCTION", "STAGE", "TASK", "PROCEDURE"] :
+        object_type => [{
+          privileges = [
+            object_type == "TABLE" ? join(",\n", ["SELECT", "INSERT", "UPDATE", "DELETE", "TRUNCATE", "REFERENCES", "REBUILD"]) :
+            object_type == "DYNAMIC TABLE" ? join(",\n", ["SELECT"]) :
+            object_type == "EXTERNAL TABLE" ? join(",\n", ["SELECT", "REFERENCES"]) :
+            object_type == "VIEW" || object_type == "MATERIALIZED VIEW" ? join(",\n", ["SELECT", "REFERENCES"]) :
+            object_type == "FILE FORMAT" ? join(",\n", ["USAGE"]) :
+            object_type == "FUNCTION" ? join(",\n", ["USAGE"]) :
+            object_type == "STAGE" ? join(",\n", ["USAGE", "READ", "WRITE"]) :
+            object_type == "TASK" ? join(",\n", ["MONITOR", "OPERATE"]) :
+            join(",\n", ["USAGE"])
+          ]
+          on_all = true
+        }]
+      }
     }
-    transformer = {
-      schema_grants            = ["CREATE TEMPORARY TABLE", "CREATE TAG", "CREATE PIPE", "CREATE PROCEDURE", "CREATE MATERIALIZED VIEW", "USAGE", "CREATE TABLE", "CREATE FILE FORMAT", "CREATE STAGE", "CREATE TASK", "CREATE FUNCTION", "CREATE EXTERNAL TABLE", "CREATE SEQUENCE", "CREATE VIEW", "CREATE STREAM"]
-      table_grants             = ["SELECT", "INSERT", "UPDATE", "DELETE", "TRUNCATE", "REFERENCES", "REBUILD"]
-      dynamic_table_grants     = ["ALL PRIVILEGES"]
-      external_table_grants    = ["SELECT", "REFERENCES"]
-      view_grants              = ["SELECT", "REFERENCES"]
-      materialized_view_grants = ["SELECT", "REFERENCES"]
-      file_format_grants       = ["USAGE"]
-      function_grants          = ["USAGE"]
-      stage_grants             = ["USAGE", "READ", "WRITE"]
-      task_grants              = ["MONITOR", "OPERATE"]
-      procedure_grants         = ["USAGE"]
+    readonly = {
+      schema_grants = [{
+        privileges              = ["USAGE"]
+        all_schemas_in_database = true
+      }]
+      schema_objects_grants = {
+        for object_type in ["TABLE", "DYNAMIC TABLE", "EXTERNAL TABLE", "VIEW", "MATERIALIZED VIEW", "FILE FORMAT", "FUNCTION", "STAGE", "TASK"] :
+        object_type => [{
+          privileges = [
+            object_type == "TABLE" || object_type == "DYNAMIC TABLE" ? join(",\n", ["SELECT"]) :
+            object_type == "EXTERNAL TABLE" || object_type == "VIEW" || object_type == "MATERIALIZED VIEW" ? join(",\n", ["SELECT", "REFERENCES"]) :
+            object_type == "FILE FORMAT" || object_type == "FUNCTION" || object_type == "STAGE" ? join(",\n", ["USAGE"]) :
+            object_type == "TASK" ? join(",\n", ["MONITOR"]) :
+            tostring([])
+          ]
+          on_all = true
```

Comment:
will we be able to override default value for `on_all = true` for specific object_type while using this compact `for` loop?

Reply:
yes, eg:
```
+ transformer     = {
[...]
  + "EXTERNAL TABLE"    = [
      + {
          + all_privileges    = true
          + object_name       = null
          + on_all            = true
          + on_future         = true
          + privileges        = null
          + schema_name       = null
          + with_grant_option = true
        },
    ]
  + "FILE FORMAT"       = [
      + {
          + all_privileges    = true
          + object_name       = null
          + on_all            = false
          + on_future         = true
          + privileges        = null
          + schema_name       = null
          + with_grant_option = false
        },
    ]
```